packages/compass-e2e-tests/helpers/commands/add-collection.ts

```diff
@@ -68,20 +68,38 @@ export async function addCollection(
     for (const [key, value] of Object.entries(
       collectionOptions.customCollation
     )) {
+      const valStr = value.toString();
+
       await browser.clickVisible(
         Selectors.createCollectionCustomCollationFieldButton(key)
       );
-      const menu = await browser.$(
+      const menu = browser.$(
```

Comment:
Why we don't await here anymore?

Reply:
browser.$() just wraps the selector into an object you can then chain. so things that actually check things or perform some action on that element only really need the await. But we're not consistent about it at all. See below in this same file where he does `const button = await browser.$(` 😄 

The documentation uses await, so that's probably better: https://webdriver.io/docs/api/browser/$

Reply:
From the docs:

> Note: only use these element objects if you are certain they still exist on the page, e.g. using the isExisting command. WebdriverIO is unable to refetch them given that there are no selector information available.

This script does a bunch of async steps, some of which are happening in a loop with waitUntil, I'd prefer webdriver to always fetch the element before running operations on it instead of failing unexpectedly with "element doesn't exist". This was the flaky part of the test, so I want to make sure it's more stable than before. Other parts of the script here didn't fail, so they are not changed. I don't want to pull more unneccessary changes to this PR. Would you be okay with having a ticket to refactor this later?

Reply:
I'm not actually sure if you do:

```javascript
const menuChainablesomething = browser.$(menuSelector)
const menuElement = await menuChainablesomething;
```

Is the element now cached on `menuChainablesomething` or only on `menuElement`? Just thinking out load. This kind of ambiguity (menuChainablesomething is both a promise and.. something?) is why I usually try and not keep that object around. But I don't think I've been very consistent about it. We definitely cache element objects in various places where we shouldn't. I've tried to reuse a selector before or wrap getting the menuElement (or that other intermediate promise-like object) in a get function. Maybe not necessary.

Reply:
side-note: I don't find their documentation very clear here. There are two types in play: `ChainablePromiseElement<WebdriverIO.Element>` and `WebdriverIO.Element` and I think it could be clearer.

Reply:
Just to make sure the context didn't get lost in private messages, I did a test and indeed stuff changed over time and the chainable caches the values after initial call. The wdio documentation [still suggests to not await on element before using commands](https://github.com/webdriverio/webdriverio/blob/main/website/docs/BestPractices.md#lazy-loading-and-promise-chaining), but for different reasons: mostly just due to this being unneccessary